crates/router/src/core/payments.rs

```diff
@@ -1993,7 +1997,7 @@ where
 {
     // On confirm is false and only router related
     let payment_data = if !is_operation_confirm(operation) {
-        let (_operation, payment_method_data) = operation
+        let (_operation, payment_method_data, pm_id) = operation
```

Comment:
This can be expanded using the [destructuring syntax.](https://doc.rust-lang.org/rust-by-example/flow_control/match/destructuring/destructure_structures.html)